android/src/main/java/com/tailscale/ipn/MainActivity.kt

```diff
@@ -182,12 +186,38 @@ class MainActivity : ComponentActivity() {
     }
   }
 
-  private fun login(url: String) {
-    // (jonathan) TODO: This is functional, but the navigation doesn't quite work
-    // as expected.  There's probably a better built in way to do this.  This will
-    // unblock in dev for the time being though.
-    val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
-    startActivity(browserIntent)
+  private fun login(urlString: String) {
+    lifecycleScope.launch {
```

Comment:
It's probably better to use our application scope here in case the activity gets destroyed but the app doesn't.

Reply:
App.getApplication().applicationScope.launch

Reply:
good point - the activity gets destroyed and recreated if the screen is rotated, and we want this to still work